pandas/core/groupby/generic.py

```diff
@@ -142,8 +130,8 @@ def pinner(cls):
 class SeriesGroupBy(GroupBy):
     _apply_whitelist = base.series_apply_whitelist
 
-    def _iterate_slices(self) -> Iterable[Tuple[Optional[Hashable], Series]]:
-        yield self._selection_name, self._selected_obj
```

Comment:
I don't know anything about this code, but: according to the property just below `self._selection_name` is not always equal to `self.obj.name` (which is what you replaced this with below)

Reply:
Yah its a bit roundabout, but it turns out that for the SeriesGroupBy they are equivalent

Reply:
OK!